barcode/templates/barcode_client.html

```diff
@@ -0,0 +1,246 @@
+<script>
+
+  var BADGE_NUM_CHAR = '~';
+  var RE_UUID = /([0-9a-f]{8}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{12})/i;
+
+  var getBadgeNumFromBarcode = function(barcode, callback) {
+    callback = callback || function() {};
+
+    // Bypass decryption if it's a number.
+    if ($.isNumeric(barcode)) {
+      callback({barcode: barcode, badge_num: barcode});
+      return;
+    }
+
+    $.ajax({
+      method: 'POST',
+      url: '../barcode/get_badge_num_from_barcode',
+      data: {
+        barcode: barcode,
+        csrf_token: csrf_token
+      },
+      success: function (response, status) {
+        if (response['badge_num'] === -1) {
+          response['error'] = response['message'];
+        }
+        response['barcode'] = barcode;
+        callback(response);
+      },
+      error: function (response, status, statusText) {
+        callback({
+          barcode: barcode,
+          error: 'Could not decode barcode: ' + statusText
+        });
+      }
+    });
+  };
+
+  var isScannerDetecting = false;
+
+  var scannerDetectionBadgeNumHandlers = [];
+  var scannerDetectionUUIDHandlers = [];
+  var scannerDetectionOnCompleteHandlers = [];
+  var scannerDetectionOnErrorHandlers = [];
+  var scannerDetectionOnReceiveHandlers = [];
+
+  $(function () {
+    $(document).scannerDetection({
+      timeBeforeScanTest: 100,
+      avgTimeByChar: 30,
+      endChar: [9, 13, '\\'],
+      startChar: [BADGE_NUM_CHAR, '\\'],
```

Comment:
I just realized, will this cause a problem for scanning UUIDs, which don't have the ~ character prepended like badge number barcodes do? If we have a scanner that doesn't add any characters, will it fail to detect a scan? 😰

Reply:
For the UUID fields, we are just treating the scanner like a keyboard. We're not doing the fancy thing where we hide the raw data from the textfield like we do with badge numbers. It still detects the scan using the timing based code (which I confirmed works on an ancient reg laptop).